pkg/health/client/client.go

```diff
@@ -158,6 +158,23 @@ func GetPathConnectivityStatusType(cp *models.PathStatus) ConnectivityStatusType
 	return status
 }
 
+func SummarizePathConnectivityStatusType(cps []*models.PathStatus) ConnectivityStatusType {
+	status := ConnStatusReachable
+	for _, cp := range cps {
+		switch GetPathConnectivityStatusType(cp) {
+		case ConnStatusUnreachable:
+			// If any status is unreachable, return it immediately.
+			return ConnStatusUnreachable
+		case ConnStatusUnknown:
+			// If the status is unknown, prepare to return it. It's
+			// going to be returned if there is no unreachable
+			// status in next iterations.
+			status = ConnStatusUnknown
+		}
+	}
+	return status
+}
```

Comment:
Arguably maybe given the amount of similar code between this and the callee, we should just fold the two functions together so that the individual path status can just return (eg in the case of an unreachable peer). But not a big deal.

Reply:
Yeah, unfortunately `GetPathConnectivityStatusType` is still needed by other packages to determine the status of individual paths, so I cannot fold it in.

I experimented with factoring out the loop body, e.g. have `foldStatus`, but I have not found it to be more readable either (due to the early return). I'll leave it as is for now.